jenkins/worker.py

```diff
@@ -0,0 +1,223 @@
+import argparse
+import datetime
+import functools
+import json
+from os.path import basename
+from pprint import pprint
+import re
+import subprocess
+import sys
+
+from google.cloud import bigquery
+from google.cloud import datastore
+
+
+# Defines a pattern for regular expression matching.
+class Pattern:
+    def __init__(self, pattern):
+        self.regex =  re.compile(pattern, re.MULTILINE)
+
+    # Returns a dictionary of named capture groups to extracted output. Returns an empty dict of no match was found.
+    def extract(self, input):
+        match = self.regex.search(input)
+        if match == None:
+            return {}
+        return match.groupdict()
+
+
+# Pattern matchers for the various fields of the '/usr/bin/time -v' output
+USER_TIME_PATTERN = Pattern(r'^\s*User time \(seconds\): (?P<user_time>\d+.\d+|\d+)')
+SYSTEM_TIME_PATTERN = Pattern(r'^\s*System time \(seconds\): (?P<system_time>\d+.\d+|\d+)')
+WALL_TIME_PATTERN = Pattern(r'^\s*Elapsed \(wall clock\) time \(h:mm:ss or m:ss\): ((?P<hours>\d{1,2}):|)'
+                            r'(?P<minutes>\d{1,2}):(?P<seconds>\d{2}\.\d{2})')
+MAX_RES_SET_SIZE_PATTERN = Pattern(r'^\s*Maximum resident set size \(kbytes\): (?P<max_set_size>\d+)')
+
+
+# Pattern matcher for extracting the HEAD commit SHA-1 hash.
+GIT_SHA1_PATTERN = Pattern(r'^(?P<sha1>[0-9a-f]{40})\s+HEAD')
+
+
+# Returns the SHA-1 hash of the commit tagged HEAD.
+def get_tot_git_sha1():
```

Comment:
google style guide: CapitalNames. Also always below.
https://google.github.io/styleguide/pyguide.html

Reply:
This changed. function_name() is ok.